subsys/bluetooth/common/dummy.c

```diff
@@ -29,12 +29,26 @@ BUILD_ASSERT(CONFIG_BT_HCI_TX_PRIO < CONFIG_BT_RX_PRIO);
 BUILD_ASSERT(CONFIG_BT_CTLR_RX_PRIO < CONFIG_BT_HCI_TX_PRIO);
 #endif /* CONFIG_BT_CTLR */
 
-/* Immediate logging is not supported with the software-based Link Layer
- * since it introduces ISR latency due to outputting log messages with
- * interrupts disabled.
+
+#if defined(CONFIG_LOG_BACKEND_UART) ||                        \
+	defined(CONFIG_LOG_BACKEND_NATIVE_POSIX) ||                \
+	defined(CONFIG_LOG_BACKEND_XTENSA_SIM)  ||                 \
+	defined(CONFIG_LOG_BACKEND_NET) ||                         \
+	(defined(CONFIG_LOG_BACKEND_RTT) &&                        \
+		(defined(CONFIG_SEGGER_RTT_MODE_BLOCK_IF_FIFO_FULL) || \
+		defined(CONFIG_LOG_BACKEND_RTT_MODE_BLOCK) ||          \
+		defined(CONFIG_LOG_IMMEDIATE_CLEAN_OUTPUT)))
```

Comment:
Looks like CONFIG_LOG_IMMEDIATE_CLEAN_OUTPUT is on the wrong parenthis level here.
It's && with RTT, but I don't think that is right.

Reply:
I think it's good, all other backend are not compatible not matter what.
RTT is not compatible if used with `CONFIG_SEGGER_RTT_MODE_BLOCK_IF_FIFO_FULL` or `CONFIG_LOG_BACKEND_RTT_MODE_BLOCK` or `CONFIG_LOG_IMMEDIATE_CLEAN_OUTPUT`

Reply:
I guess we could also say that any back-end with CONFIG_LOG_IMMEDIATE_CLEAN_OUTPUT are not compatible. I can change that if you prefer. The result will be the same tough

Reply:
Yes, that would be preferable.